minidump-processor/src/op_analysis.rs

```diff
@@ -48,30 +48,110 @@ pub enum OpAnalysisError {
+    pub instruction_pointer_update: Option<InstructionPointerUpdate>,
     /// A list of all registers which were used by this instruction.
     pub registers: BTreeSet<&'static str>,
 }
 
+/// A list of booleans representing whether this instruction could have caused
+/// a particular type of crash
+/// Note that memory access crashses are checked through `memory_accesses`
+// TODO: remove `is_common_memory_crash_instruction` field once `yaxpeax` provides access types
+// for operands of all instructions
+#[derive(Clone, Debug)]
+pub struct PossibleCrashInfo {
+    pub is_common_memory_crash_instruction: bool,
+    pub int_division_by_zero: bool,
+    pub priv_instruction: bool,
+}
+
 /// Details about a memory access performed by an instruction
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
 pub struct MemoryAccess {
-    /// The address of the memory access
-    pub address: u64,
-    /// Whether or not this memory access is likely the result of a null-pointer dereference
-    pub is_likely_null_pointer_dereference: bool,
-    /// Whether or not this memory access was part of a likely guard page.
-    pub is_likely_guard_page: bool,
+    // Information
+    pub address_info: MemoryAddressInfo,
     /// The size of the memory access
     ///
     /// Note that this is optional, as there are weird instructions that do not know the size
     /// of their memory accesses without more complex context.
     pub size: Option<u8>,
+    /// The type of the memory access
+    pub access_type: MemoryAccessType,
+}
+
+/// Details about update of instruction pointer performed by an instruction
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct InstructionPointerUpdate {
+    /// Information about the address that instruciton pointer is being updated to
+    pub address_info: MemoryAddressInfo,
+}
+
+/// Details about a memory address of a memory access or an instruction pointer update
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct MemoryAddressInfo {
+    /// The address
+    pub address: u64,
+    /// Whether or not this memory address is likely the result of a null-pointer dereference
+    pub is_likely_null_pointer_dereference: bool,
+    /// Whether or not this memory address was part of a likely guard page.
+    pub is_likely_guard_page: bool,
+}
+
+/// The direction of a memory access
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum MemoryAccessType {
+    Read,
+    Write,
+    ReadWrite,
+
+    // TODO: Remove this variant once `yaxpeax` is used to derive access type of every operand
+    UncommonInstructionAccess,
+}
+
+pub trait MemoryAccessVecExt {
+    fn contains_access(&self, address: u64, access_type: MemoryAccessType) -> bool;
+}
+
+impl MemoryAccessVecExt for Vec<MemoryAccess> {
+    fn contains_access(&self, address: u64, access_type: MemoryAccessType) -> bool {
+        self.iter().any(|access| {
+            let size = access.size.unwrap_or(64) as u64;
```

Comment:
I'm going to go out on a limb here and say that you were probably thinking _bits_, but that number is in **bytes**. That's a pretty big memory access for one instruction! 😂

As for how to handle this -- Just `return false`. There's only a few rare, super-weird instructions on x64 that don't know the size of memory they access, and we don't really have to care about them.